### ISSUE TITLE
no_proxy is inconsistent with urllib and makes mixing libraries HAVE to put . when not needed if this library is not in use.

### ISSUE THREAD

**Issue by MikeMoore63** [websocket-client/websocket-client#968]
I have read issue #700 and read the  argument the ucrrent implementation is "more flexible". Having been burnt with this just recently I would argue the inconsistency is worse.

If you use a black box application that includes requests and then adds websocket support using this library things go horribly wrong. I eventually traced the issues to the nigh on unique approach websockets takes to handling no_proxy.

Appreciate its inconsistent please read  https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/ for more on the issues.

Please read the "lowest common denominator" which unfortunately the websocket-client does not meet. I was religously following this.

Googles gcloud supports proxies and I had been using it for years with no_proxy with domains with no ".". All was well until 2 weeks ago when i started using a new feature for tunnelling that uses this client. This was the only feature that did not work. It took me debugging the gcloud to eventually work out this library was even part of gcloud and its has this apparently nigh on unique implementation of no_proxy. 

I would expect this to simply be consistent with all other python implementations it is currently consistent with curl but not wget. So consistent with urllib https://github.com/python/cpython/blob/030a713183084594659aefd77b76fe30178e23c8/Lib/urllib/request.py#L2548 

```python
# check if the host ends with any of the DNS suffixes
    for name in no_proxy.split(','):
        name = name.strip()
        if name:
            name = name.lstrip('.')  # ignore leading dots
            name = name.lower()
            if hostonly == name or host == name:
                return True
            name = '.' + name
            if hostonly.endswith(name) or host.endswith(name):
                return True
```

vs

https://github.com/websocket-client/websocket-client/blob/a394b46c560cb78672f95bbb3836305e8511d1b9/websocket/_url.py#L126

```python
    if not no_proxy:
        if v := os.environ.get("no_proxy", os.environ.get("NO_PROXY", "")).replace(
            " ", ""
        ):
            no_proxy = v.split(",")
    if not no_proxy:
        no_proxy = DEFAULT_NO_PROXY_HOST

    if "*" in no_proxy:
        return True
    if hostname in no_proxy:
        return True
    if _is_ip_address(hostname):
        return any(
            [
                _is_address_in_network(hostname, subnet)
                for subnet in no_proxy
                if _is_subnet_address(subnet)
            ]
        )
    for domain in [domain for domain in no_proxy if domain.startswith(".")]:
        if hostname.endswith(domain):
            return True
```
instead the "." should be removed and checks done against endwith for consistencies sake.

**Comment by engn33r** [comment]
@MikeMoore63 To make sure I understand, can you confirm that this is that change you are suggesting in [_url.py](https://github.com/websocket-client/websocket-client/blob/2507a7023c00807e6d89d474b308cc6f8e2bb280/websocket/_url.py#L126) for alignment with urllib?

```diff
for domain in [domain for domain in no_proxy if domain.startswith(".")]:
-    if hostname.endswith(domain):
+    endDomain = domain.lstrip('.'):
+    if hostname.endswith(endDomain):
        return True
```

**Comment by MikeMoore63** [comment]
Correct

**Comment by engn33r** [comment]
Thank you for confirming. The change makes sense to me, thanks for reporting this. It is interesting that the GitLab post you linked to shows wget and Go don't strip the leading dot and therefore don't meet the lowest common denominator suggestions at the end of that post.

Fixed in b02f08f251f58a4a9acaf85d6858147f4af70fda and the update will get into the next release.

**Comment by MikeMoore63** [comment]
Ok I still think you have an issue because of the if on the end
```
for domain in [domain for domain in no_proxy if domain.startswith(".")]:
```

at line 126 thinking about it this is the bigger issue. You do need lstrip if you removed this.

The point is a "." is optional.
